include/zfs_fletcher.h

```diff
@@ -35,11 +35,20 @@ extern "C" {
 
 /*
  * fletcher checksum functions
+ *
+ * Note: Fletcher checksum methods expect buffer size to be 4B aligned. This
+ * limitation stems from the algorithm design. Performing incremental checksum
+ * without said alignment would yield different results. Therefore, the code
+ * includes assertions for the size alignment.
+ * For compatibility, it is required that some code paths calculate checksum of
+ * potentially unknown buffer sizes. For this purpose, unsafe version of
+ * checksum method is added. This method will ignore last (size % 4) bytes of
+ * buffer.
  */
-
 void fletcher_2_native(const void *, uint64_t, zio_cksum_t *);
 void fletcher_2_byteswap(const void *, uint64_t, zio_cksum_t *);
 void fletcher_4_native(const void *, uint64_t, zio_cksum_t *);
+void fletcher_4_native_unsafe(const void *, uint64_t, zio_cksum_t *);
```

Comment:
yeah, definitely... it does not describe it correctly either. I'm terrible at naming stuff :) Any suggestions? (_compat)?